src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -680,25 +677,30 @@ class WelcomePage extends Disposable {
 		return result;
 	}
 
+	private called: boolean;
 	private async mapListEntries(recents: (IRecentWorkspace | IRecentFolder)[], fileService: IFileService, container: HTMLElement): Promise<HTMLElement[]> {
-		const result: HTMLElement[] = [];
-		for (let i = 0; i < recents.length; i++) {
-			const recent = recents[i];
-			let relativePath: string;
-			let fullPath: URI;
-			let windowOpenable: IWindowOpenable;
-			if (isRecentFolder(recent)) {
-				windowOpenable = { folderUri: recent.folderUri };
-				relativePath = recent.label || this.labelService.getWorkspaceLabel(recent.folderUri, { verbose: true });
-				fullPath = recent.folderUri;
-			} else {
-				relativePath = recent.label || this.labelService.getWorkspaceLabel(recent.workspace, { verbose: true });
-				windowOpenable = { workspaceUri: recent.workspace.configPath };
+		if (!this.called) {
```

Comment:
I think we don't need this 'called' flag, if we clear the items before rendering the recent list we should be good.